src/Backend/OnboardingInsuranceAPI/Areas/Insurance/AddInsurance.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using OnboardingInsuranceAPI.ErrorHandling;
+using OnboardingInsuranceAPI.Services;
+
+namespace OnboardingInsuranceAPI.Areas.Insurance;
+
+public class AddInsurance: IHandler
+{
+    private readonly DataContext _context;
+    private readonly ILogger<AddInsurance> _logger;
+
+    public AddInsurance(DataContext context, ILogger<AddInsurance> logger)
+    {
+        _context = context;
+        _logger = logger;   
+    }
+
+    public async Task Handle(InsuranceData item)
+    {
+        if (string.IsNullOrEmpty(item.Pid))
+        {
+            _logger.LogWarning("Error, received pid is empty!");
+            throw new ApiException(ErrorCode.InvalidQueryParameters);
+        }
+
+        var user = await _context.Users.FirstOrDefaultAsync(u => u.Pid == item.Pid);
+
+        if (user is null)
+        {
+            _logger.LogWarning("Error reading user, user is null!");
+            throw new ApiException(ErrorCode.NotFound);
+        }
+
+        user.CarInsurance = item.CarInsurance;
```

Comment:
This could be problematic. Because you are saying by this class "ADD insurance" but you are actually doing is Updating the insurance. It doesn't matter if user already has insurance or not.

Reply:
I can fix it on frontend, when insurance already added I'll hide button save ;)